### PR TITLE
Update contributing guide

### PR DESCRIPTION
**Pull request by purva-thakre**
## Changes
Notable changes that this PR has either accomplished or will accomplish. Feel free to add more lines to the itemized list
below.

  -  Corrects the formatting for docstrings parameters
  - replaces mentions of sphinx-wagtail theme by furo
  - adds sphinx copybutton

## Checklist
Before marking your PR ready for review, make sure you checked the following locally. If this is your first PR, you might be notified of some workflow failures after a maintainer has approved the workflow jobs to be run on your PR. 

Additional information is available in the [documentation](https://toqito.readthedocs.io/en/latest/contributing.html#testing).

  -  [ ] Use `ruff` for errors related to code style and formatting.
  -  [ ] Verify all previous and newly added unit tests pass in `pytest`.
  -  [ ] Check the documentation build does not lead to any failures. `Sphinx` build can be checked locally for any failures related to your PR
  -  [ ] Use `linkcheck` to check for broken links in the documentation
  -  [ ] Use `doctest` to verify the examples in the function docstrings work as expected.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -43,6 +43,7 @@
     "sphinx.ext.githubpages",
     "sphinx.ext.napoleon",
     "sphinxcontrib.bibtex",
+    "sphinx_copybutton",
```
Comment (purva-thakre) [inline docs/conf.py:46]: This is a new dependency which lets a user copy a codeblock in the documentation.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
 ## Description
-Provide a brief description of the PR's purpose here. If your PR is supposed to fix an existing issue, use
-a [keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) to link your PR to the issue.
+<!--Provide a brief description of the PR's purpose here. If your PR is supposed to fix an existing issue, use
```
Comment (vprusso) [inline .github/PULL_REQUEST_TEMPLATE.md:2]: Is there a reason we are flanking these with the `<!--` HTML comment syntax? Might just not be aware of why this change is being made.

**Review by vprusso — APPROVED** [review]
Only a minor question, but this is great. Excellent updates, @purva-thakre !

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (purva-thakre) [reply .github/PULL_REQUEST_TEMPLATE.md:2]: I wanted to hide the suggestions in `.md` format. Otherwise, if a PR is created, it shows up like this:

image

With the newer changes, the instructions are still available if a user wants them but they don't clutter the description box.
image

Comment (vprusso) [reply .github/PULL_REQUEST_TEMPLATE.md:2]: Ah okay. In that case, everything else seems fine, LGTM! 🚀
